native/src/main/java/tech/sourced/babelfish/DriverResponse.java

```diff
@@ -38,12 +38,16 @@ void parseCode(EclipseCPPParser parser, String source) {
         translationUnit = parser.parseCPP(source);
     }
 
+    // Note: since we're using the System.out output stream with Jackson, output will
+    // start to be written before this call so its not a deterministic "send everything".
+    // The reason to not use a ByteArrayOutputStream and send everything in one go is that
+    // sometimes memory can grow too much with some files.
     void send() throws ResponseSendException {
         // FIXME: this includes the errors in the already started document
         try {
             formatWritter.writeValue(this);
-            ByteArrayOutputStream byteOut = formatWritter.getByteOutputStream();
-            System.out.write(byteOut.toByteArray());
+            OutputStream byteOut = formatWritter.getOutputStream();
```

Comment:
Nit for future consideration: It's no longer really `byteOut`.

Reply:
I kept it because the argument to OutputStream's write() methods are still bytes.